File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const numIdx = {};
    for (let i = 0; i < nums.length; i++) {
        const diff = target - nums[i];
        if (diff in numIdx) return [numIdx[diff], i];
        numIdx[nums[i]] = i;
    }
};

File: 0001-two-sum/0001-two-sum.py
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}

def two_sum(nums, target):
    num_idx = {}
    for i, num in enumerate(nums):
        diff = target - num
        if diff in num_idx:
            return [num_idx[diff], i]
        num_idx[num] = i
